java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java

```diff
@@ -315,46 +312,57 @@ private <RequestT> SdkHttpFullRequest prepareRequest(
         } catch (URISyntaxException e) {
             throw new IllegalArgumentException("Invalid request URI: " + url.toString());
         }
+
+        ContentStreamProvider bodyProvider = body != null ? ContentStreamProvider.fromInputStreamSupplier(body::getInputStream) : null;
```

Comment:
So the `body` is `OpenSearchRequestBodyBuffer` and is always backed by buffer, could we use `ContentStreamProvider::fromByteArrayUnsafe` (zero copy) instead of creating a wrapping input stream over the array?